roles/confluent.kafka_broker/tasks/dynamic_groups.yml

```diff
@@ -0,0 +1,49 @@
+---
+- name: Load server.properties
+  slurp:
+    src: "{{ kafka_broker.config_file }}"
+  register: slurped_properties
+
+# Zookeeper shell cannot read server.properties if secrets protection is enabled. Creates a non encrypted file
+- name: Create Zookeeper TLS Client Config
+  template:
+    src: zookeeper-tls-client.properties.j2
+    dest: "{{ kafka_broker.zookeeper_tls_client_config_file }}"
+    mode: 0640
+    owner: "{{ kafka_broker_user }}"
+    group: "{{ kafka_broker_group }}"
+  when:
+    - >
+      'zookeeper.ssl.client.enable = true' in slurped_properties.content|b64decode
+        or 'zookeeper.ssl.client.enable=true' in slurped_properties.content|b64decode
+    - kafka_broker_secrets_protection_enabled|bool
+
+# TODO When Zookeeper is removed, use kafka-metadata-quorum script. See KIP-595
+- name: Get Controller Broker ID
+  shell: >
+    set -o pipefail &&
+      {{ binary_base_path }}/bin/zookeeper-shell {{ groups['zookeeper'][0] }}:{{ zookeeper_client_port }}
+          {%- if 'zookeeper.ssl.client.enable = true' in slurped_properties.content|b64decode
+                or 'zookeeper.ssl.client.enable=true' in slurped_properties.content|b64decode %}
+            -zk-tls-config-file {{ kafka_broker.zookeeper_tls_client_config_file \
+              if kafka_broker_secrets_protection_enabled else kafka_broker.config_file }}
+          {%- endif %}
+          get /controller | grep brokerid
+  register: controller_query
+  run_once: true
+  changed_when: false
+  check_mode: false
+
+- name: Get broker_id
+  set_fact:
+    controller_json: "{{ controller_query.stdout }}"
+    broker_id: "{{ (slurped_properties.content|b64decode).split('\n') | select('match', '^broker.id.*') | list \
```

Comment:
we should really create a filter or something for this! look sufficient though

Reply:
It was extracted from `create_ordered_kafka_groups.yml`. I tried to change as little as possible. 😄 But yes, room for improvement.